tenzir/integration/tests/expression.bats

```diff
@@ -55,6 +55,17 @@ z = [...x, 3, ...y, ...42]
 EOF
 }
 
+@test "record indexing" {
+  echo '
+  {"x": {"x": 1}, "y": "x"}
+  {"x": {"x": "1"}, "y": "x"}
+  {"x": {"x": 1}}
+  {"x": {}, "y": "x"}
+  {"y": "x"}
+  {}' |
+    check tenzir 'read_json | z = x[y]'
```

Comment:
Have you considered this?

```
from [
  {...},
  {...},
  {...},
  {...},
]
z = x[y]
```

Reply:
You'll get type clashes in `from` that way.

Reply:
Ah yes. I actually ran into this as well!

I had issues with:
```
from [
  {x: 1, y: "foo"},
  {x: 2, y: "bar"},
  {x: 1},
  {x: 2},
  {y: "foo"},
]
taste 1
```

Reply:
That should work fine, types of `x` and `y` don't change in there. Maybe you had a slightly different example.